### PR TITLE
fix: add password requirements to user registration (#9951)

### LINKED ISSUE #9951: Password requirements on registration page
**Describe the bug**
On the registration/"Create an Account" page in Spartacus Electronics, when you enter a password that does not fulfil the requirements, it shows "This is not a valid password format" but does not tell you what a valid format looks like (see Screenshot).

**Tell us the version of Spartacus**
- Library version: Not sure. In yDC I used this template to create the instance: sap_commerce_2005-cx

**To Reproduce**
1. Go to Spartacus Electronics Homepage
2. Click on "Sign In / Register"
3. Click on "Register"
4. Scroll down to "Password"
5. Type anything into the box 
6. Error/Hint Message appears

**Expected behavior**
I would expect a message that shows me, what a valid password requires: be one lower, one upper case char, a number AND a special char.

**Screenshots**
Screenshot 2020-11-19 at 16 06 51

**Desktop (please complete the following information):**
 - OS: macOS Catalina
 - Browser: Chrome
 - Version: 86.0.4240.198

### PR DESCRIPTION
**Pull request by I539321**
Closes https://github.com/SAP/spartacus/issues/9951

### CODE REVIEW

**Comment by I539321** [comment]
Hi @bit172.

This is the new PR from a repo. branch instead of that of a fork.

Before I proceed with making the changes to e2e, can you confirm whether the code is on the right track?

**Comment by bit172** [comment]
> Hi @bit172.
> 
> This is the new PR from a repo. branch instead of that of a fork.
> 
> Before I proceed with making the changes to e2e, can you confirm whether the code is on the right track?

Looks like it's on the right track!

**Comment by I539321** [comment]
I initially thought the failing tests were due to the changes pushed here, but that does not seem to be the case.

Nothing else to add here from my side, feel free to proceed with a final review.

In the meanwhile I'll create the PR for adding the explanation for the password requirements to the Spartacus docs.

**Comment by bit172** [comment]
> I initially thought the failing tests were due to the changes pushed here, but that does not seem to be the case.
> 
> Nothing else to add here from my side, feel free to proceed with a final review.
> 
> In the meanwhile I'll create the PR for adding the explanation for the password requirements to the Spartacus docs.

Right, someone is fixing e2e tests so don't worry about that. As long as your own test are passing that's fine

**Comment by I539321** [comment]
Hi @Xymmer.

Can you please check if the default requirements [added here](https://github.com/SAP/spartacus/pull/15490/files#diff-dea96abe6821e1940a617c48ed48e9985f92eaa09b0566e25a1dd10aa6a591c3R175) are appropriate? They are a reflection of the first bulleted list from [here](https://help.sap.com/viewer/9433604f14ac4ed98908c6d4e7d8c1cc/2105/en-US/d7c5a42f7d024e37991a95709eb149e5.html).

Let me know if anything should be changed.

**Comment by RadhepS** [comment]
Thanks for working on this, it's a really important change!

I think once you get the password requirements down, you can sync with @gladius-mtl to make sure that the wording for the warning message is fine. (Hope you're the right person to check this out Matthew haha)

File: projects/assets/src/translations/en/common.ts [hunk 1]
```diff
@@ -171,6 +171,8 @@ export const common = {
     required: 'This field is required',
     cxInvalidEmail: 'This is not a valid email format',
     cxInvalidPassword: 'This is not a valid password format',
+    cxPasswordRequirements:
+      'Passwords must include: both uppercase and lowercase letters, one or more numerical digits, one or more special characters, at least 8 total characters', // change me to reflect the current password requirements
```
Comment (RadhepS) [inline projects/assets/src/translations/en/common.ts]: Thanks for working on this, it's a really important change!

I think once you get the password requirements down, you can sync with @gladius-mtl to make sure that the wording for the warning message is fine. (Hope you're the right person to check this out Matthew haha)

Comment (gladius-mtl) [reply projects/assets/src/translations/en/common.ts]: I propose the following text:

`Passwords must include at least 8 characters, a mixture of both uppercase and lowercase letters, at least one number, and at least one special character, such as !@#?$`

Comment (I539321) [reply projects/assets/src/translations/en/common.ts]: Password requirement message changed accordingly!

**Comment by RadhepS** [comment]
Can we make changes to the e2e tests that includes a check for this?
As mentioned here: https://github.com/SAP/spartacus/pull/15490#issuecomment-1083106617
Also, can we add a unit test to cover this case as well?

**Review by RadhepS — CHANGES_REQUESTED** [review]
Can we make changes to the e2e tests that includes a check for this?
As mentioned here: https://github.com/SAP/spartacus/pull/15490#issuecomment-1083106617
Also, can we add a unit test to cover this case as well?

**Pull request closed without merging**